Library/Homebrew/dependency_collector.rb

```diff
@@ -144,12 +144,12 @@ def parse_class_spec(spec, tags)
   end
 
   def resource_dep(spec, tags)
-    tags << :build << :test
+    tags << :build << :test << :fetch
     strategy = spec.download_strategy
 
     if strategy <= HomebrewCurlDownloadStrategy
-      brewed_curl_dep_if_needed(tags)
       parse_url_spec(spec.url, tags)
+      brewed_curl_dep_if_needed(tags)
```

Comment:
There's a bug here where only the last line of the `if` statement can return a `Dependency`. If a URL were marked with `:homebrew_curl` and ending in ".rar", two dependencies would need to be added, but not sure how to rewrite this to handle that.

Before:
```
$ brew deps --tree --include-build libao
libao
├── autoconf
│   └── m4
├── automake
│   └── autoconf
│       └── m4
├── libtool
│   └── m4
└── pkg-config
```

After:
```
$ brew deps --tree --include-build libao
libao
├── curl
│   ├── pkg-config
│   ├── brotli
│   │   └── cmake
│   ├── libidn2
│   │   ├── pkg-config
│   │   ├── gettext
│   │   └── libunistring
│   ├── libnghttp2
│   │   └── pkg-config
│   ├── libssh2
│   │   └── openssl@1.1
│   │       └── ca-certificates
│   ├── openldap
│   │   └── openssl@1.1
│   │       └── ca-certificates
│   ├── openssl@1.1
│   │   └── ca-certificates
│   ├── rtmpdump
│   │   └── openssl@1.1
│   │       └── ca-certificates
│   └── zstd
│       └── cmake
├── autoconf
│   └── m4
├── automake
│   └── autoconf
│       └── m4
├── libtool
│   └── m4
└── pkg-config
```

Reply:
Refactor to be called `resource_deps` and return an Array?